ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -176,7 +176,7 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
         try {
             IngestionResult ingestionResult;
             if (streamSourceInfo.getStream() == null || streamSourceInfo.getStream().available() <= 0) {
-                throw new IngestionClientException("Stream");
+                throw new IngestionClientException("Stream is null or empty");
```

Comment:
Consider splitting and throwing two distinct errors - "Null stream received" and "Empty stream received"